escrow/license_contract.proto

```diff
@@ -0,0 +1,184 @@
+syntax = "proto3";
+
+package escrow;
+import "google/protobuf/timestamp.proto";
+
+option java_package = "io.singularitynet.daemon.License";
+
+service Training {
+
+}
+
+service LicenseContract {
+
+    rpc CreateLicense (LicenseCreateRequest) returns (LicenseDataResponse) {
+    }
+    //You need to pass the ChannelId
+    rpc GetLicenseForChannel (LicenseReadRequest) returns (LicenseDataResponse) {
+    }
+    //this is to renew the License
+    rpc RenewLicense (LicenseUpdateRequest) returns (LicenseDataResponse) {
+    }
+
+    rpc CancelLicense (LicenseUpdateRequest) returns (LicenseDataResponse) {
+    }
+    //All license associated with this user will be sent back
+    rpc GetAllLicensesForUser (CallerAuthentication) returns (AllLicensesResponse) {
+    }
+    //this is to only tell you the types of licenses supported by provider , we will start with subscription
+    //and then scale this up to Tier Pricing
+    rpc GetLicensesSupportedByProvider (LicenseProviderReadRequest) returns (LicenseTypesSupportedResponse) {
+    }
+
+    //filter by licenses created for given service Id
+    rpc GetAllLicenseByServiceIds (LicenseProviderReadRequest) returns (AllLicensesResponse) {
+    }
+
+    rpc GetLicenseUsage (LicenseReadRequest) returns (UsageResponse) {
+    }
+
+    //To Be called By Daemon ONLY, Daemons's address is listed in service metadata, this will be used by the licensing server
+    //to check for authorizations on Signature.
+    rpc CheckEligibilityAndIncrementUsage (UpdateLicenseUsageRequest) returns (UsageResponse) {
+    }
+
+    //To Be called By Daemon ONLY, Daemons's address is listed in service metadata, this will be used by the licensing server
+    //    //to check for authorizations on Signature.
+    rpc DecrementUsage (UpdateLicenseUsageRequest) returns (UsageResponse) {
+    }
+
+
+}
+
+
+message CallerAuthentication {
+    //Signature will compromise the following data:
+    // (_method name, current block, user address) signed by private key
+    // Example ("_GetLicense", "block_number") OR ("_CreateLicense", "block_number")
+    bytes signature = 1;
+    //current block number (signature will be valid only for short time around this block number)
+    uint64 current_block = 2;
+}
+
+message UpdateLicenseUsageRequest {
+    CallerAuthentication auth = 1;
+    uint64 channel_id = 2;
+    string service_id = 3;
+
+}
+
+message LicenseCreateRequest {
+    CallerAuthentication auth = 1;
+    //the channel exclusively created for this license needs to be passed
+    uint64 channel_id = 2;
```

Comment:
So, am I right we are choosing channel per license approach?

Reply:
yes , one channel per license